### PR TITLE
Cloud awareness

### PR DESCRIPTION
**Pull request by nilsanderselde**
Change `databaseAccount` to `databaseHostname` so that the correct domain for the current Azure cloud environment can be used.

Needed to allow ARO RP to be cloud aware.

### CODE REVIEW

File: pkg/gencosmosdb/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -110,7 +113,16 @@ func (c *databaseClient) do(ctx context.Context, method, path, resourceType, res
 }
 
 func (c *databaseClient) _do(ctx context.Context, method, path, resourceType, resourceLink string, expectedStatusCode int, in, out interface{}, headers http.Header) (*http.Response, error) {
-	req, err := http.NewRequestWithContext(ctx, method, "https://"+c.databaseAccount+".documents.azure.com/"+path, nil)
```
Comment (jim-minter) [inline pkg/gencosmosdb/cosmosdb/cosmosdb.go:113]: how about let's change `databaseAccount` to `databaseHostname` and do all the environment logic outside the library?

**Review by jim-minter — CHANGES_REQUESTED** [review]

File: example/example_test.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ func TestE2E(t *testing.T) {
 	t.Logf("%#v\n", docs)
 
 	i := dc.ChangeFeed(nil)
-	docs, err = i.Next(ctx)
+	docs, err = i.Next(ctx, 1)
```
Comment (nilsanderselde) [inline example/example_test.go:172]: I discovered this hadn't been updated, so I got it working again.

File: pkg/gencosmosdb/bindata.go [hunk 3]
```diff
@@ -228,8 +227,8 @@ func triggerGo() (*asset, error) {
 // It returns an error if the asset could not be found or
 // could not be loaded.
 func Asset(name string) ([]byte, error) {
-	canonicalName := strings.Replace(name, "\\", "/", -1)
-	if f, ok := _bindata[canonicalName]; ok {
+	cannonicalName := strings.Replace(name, "\\", "/", -1)
```
Comment (jim-minter) [inline pkg/gencosmosdb/bindata.go:230]: please can you update locally to the latest go-bindata and regenerate?

**Review by jim-minter — CHANGES_REQUESTED** [review]
1 nit

File: example/example_test.go (hunk 2 above)
Comment (jim-minter) [reply example/example_test.go:172]: thanks!

**Pull request merged**